bicytok/figures/figure2.py

```diff
@@ -26,34 +28,36 @@ def makeFigure():
 
     # Define off-target conditions using a dictionary
     off_target_conditions = {
-        0: (CITE_DF["CellType3"] != targCell),     # All non-memory Tregs
-        1: (CITE_DF["CellType2"] != "Treg"),       # All non-Tregs
-        2: (CITE_DF["CellType3"] == "Treg Naive")  # Naive Tregs
+        0: (CITE_DF["CellType3"] != targCell),  # All non-memory Tregs
+        1: (CITE_DF["CellType2"] != "Treg"),  # All non-Tregs
+        2: (CITE_DF["CellType3"] == "Treg Naive"),  # Naive Tregs
     }
 
     # Set off_target based on offTargState
     if offTargState in off_target_conditions:
         off_target = off_target_conditions[offTargState].astype(int)
     else:
         raise ValueError("Invalid offTargState value. Must be 0, 1, or 2.")
-    
-    
+
     KL_values, EMD_values = KL_EMD_1D(markerDF, on_target, off_target)
-    
+
     top_5_KL_indices = np.argsort(KL_values)[-5:]
```

Comment:
I am setting indices of ignored receptors to NaN which is throwing off the sorting. Either you should use 
top_5_KL_indices = np.argsort(np.nan_to_num(KL_values))[-5:] to treat the nans as zeros in the sort or I should store ignored distances as something else.